index.js

```diff
@@ -44,12 +44,12 @@ var byObserver = function (Observer) {
 
 module.exports = (function () {
 	// Node.js
-	if ((typeof process === 'object') && process && (typeof process.nextTick === 'function')) {
+	if (typeof process === 'object' && typeof process.nextTick === 'function') {
 		return process.nextTick;
 	}
 
 	// MutationObserver
-	if ((typeof document === 'object') && document) {
+	if (typeof document === 'object' && document.createTextNode) {
```

Comment:
It will crash if for some weird reason `document === null` (that's why there's additional `&& document` check